File: WorldFeed/Server/src/WorldFeed.Common/DTO/Cursor/TwitterListCursorQueryResultDTO.cs
namespace WorldFeed.Common.DTO.Cursor
{
    using Newtonsoft.Json;

    using WorldFeed.Common.Public.Models.Interfaces.DTO;
    using WorldFeed.Common.Public.Models.Interfaces.DTO.QueryDTO;

    public class TwitterListCursorQueryResultDTO : BaseCursorQueryDTO<ITwitterListDTO>, ITwitterListCursorQueryResultDTO
    {
        private ITwitterListDTO[] twitterLists;

        [JsonProperty("lists")]
        public ITwitterListDTO[] TwitterLists
        {
            get => this.twitterLists ?? new ITwitterListDTO[0];
            set
            {
                this.twitterLists = value;
                Results = value;
            }
        }

        public override int GetNumberOfObjectRetrieved()
        {
            return TwitterLists.Length;
        }
    }
}

File: WorldFeed/Client/src/app/core/Core/DTO/Cursor/TwitterListCursorQueryResultDTO.ts
import {BaseCursorQueryDTO} from "./BaseCursorQueryDTO";
import {ITwitterListDTO} from "../../../Public/Models/Interfaces/DTO/ITwitterListDTO";
import {ITwitterListCursorQueryResultDTO} from "../../../Public/Models/Interfaces/DTO/QueryDTO/ITwitterListCursorQueryResultDTO";
import {TwitterListDTO} from "../TwitterListDTO";

export class TwitterListCursorQueryResultDTO extends BaseCursorQueryDTO<ITwitterListDTO> implements ITwitterListCursorQueryResultDTO {
  private _twitterLists: ITwitterListDTO[];

  // [JsonProperty("lists")]
  get twitterLists(): ITwitterListDTO[] {
    return this._twitterLists ?? new TwitterListDTO()[0];  // new ITwitterListDTO[0];
  }

  set twitterLists(value: ITwitterListDTO[]) {
    this._twitterLists = value;
    super.results = value;
  }

  public getNumberOfObjectRetrieved(): number {
    return this.twitterLists.length;
  }
}
